commit/merkleroot/rmn/controller.go

```diff
@@ -528,26 +564,37 @@ func gotSufficientObservationResponses(
 		}
 	}
 
+	resultChains := mapset.NewSet[cciptypes.ChainSelector]()
+
 	for sourceChain := range updateRequests {
-		// make sure we got at least F+1 observing the same merkle root for a target chain.
 		countsPerRoot, ok := merkleRootsCount[sourceChain]
 		if !ok || len(countsPerRoot) == 0 {
-			return false
+			lggr.Infow("chain skipped, zero roots observed", "chain", sourceChain)
+			continue
 		}
+
 		homeChainF, exists := homeFMap[cciptypes.ChainSelector(sourceChain)]
 		if !exists {
 			lggr.Errorw("no F for chain", "chain", sourceChain)
-			return false
+			continue
 		}
 
 		values := maps.Values(countsPerRoot)
 		sort.Slice(values, func(i, j int) bool { return values[i] < values[j] })
 		if consensus.LtFPlusOne(homeChainF, values[len(values)-1]) {
```

Comment:
Inferring from this code, but is this testing the F+1 on the highest count achieved for a particular root? Maybe `slices.Max` would make that clearer:

```suggestion
		mostVotedRootVotes := slices.Max(maps.Values(countsPerRoot))
		if consensus.LtFPlusOne(homeChainF, mostVotedRootVotes) {
```